core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/ProcessActivityResult.kt

```diff
@@ -38,7 +38,10 @@ data class ProcessActivityResult(
       resultCode == Activity.RESULT_OK &&
       data != null
     ) {
-      actions.offer(Filter(data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)[0]))
+      val results = data.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)
+      if (results != null && results.first() != null) {
+        actions.offer(Filter(results.first()))
+      }
```

Comment:
**Optional:** We can try to write this using let, take & other [scope functions in Kotlin](https://kotlinlang.org/docs/scope-functions.html)